sklearn/discriminant_analysis.py

```diff
@@ -72,7 +73,7 @@ def _cov(X, shrinkage=None, covariance_estimator=None):
                 s = empirical_covariance(X)
             else:
                 raise ValueError("unknown shrinkage parameter")
-        elif isinstance(shrinkage, float) or isinstance(shrinkage, int):
+        elif isinstance(shrinkage, Number):
```

Comment:
I think `Number` includes complex numbers. In this case, I think we want `Real`:

```suggestion
        elif isinstance(shrinkage, Real):
```

Reply:
Updated :)